File: CS/WindowsApplication3/Form1.cs
using DevExpress.Utils;
using DevExpress.Utils.Drawing;
using DevExpress.XtraEditors;
using DevExpress.XtraEditors.Controls;
using DevExpress.XtraEditors.ViewInfo;
using System;
using System.Data;
using System.Drawing;
using System.IO;


namespace WindowsApplication3
{
    public partial class Form1 : XtraForm
    {
        char separatorChar;

        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            carsBindingSource.DataSource = GetCarSchedulingDataTable();
            separatorChar = checkedComboBoxEdit1.Properties.SeparatorChar;
        }

        DataTable GetCarSchedulingDataTable()
        {
            DataTable table = new DataTable();
            table.TableName = "CarScheduling";
            table.Columns.Add(new DataColumn("ID", typeof(int)));
            table.Columns.Add(new DataColumn("Model", typeof(string)));
            table.Columns.Add(new DataColumn("Picture", typeof(byte[])));
            Random random = new Random();
            for(int i = 0; i < 20; i++)
            {
                int index = i + 1;
                Image image = WindowsApplication3.Properties.Resources.about_32x32;
                if(random.Next(0, 2) == 0)
                    image = WindowsApplication3.Properties.Resources.convert_32x32;
                using(var ms = new MemoryStream())
                {
                    image.Save(ms, image.RawFormat);
                    table.Rows.Add(index, "Car " + random.Next(0, 100), ms.ToArray());

                }
            }
            return table;
        }

        private string GetCheckedItemDescription(string[] checkedItems, Font font, Point point, Rectangle rect)
        {
            int[] widths = GetValuesWidths(font, checkedItems);
            Rectangle valRect = new Rectangle(rect.X, rect.Y, widths[0], rect.Height);
            int n = 0;
            while(!valRect.Contains(point) && n < widths.Length - 1)
            {
                valRect.X += widths[n++];
                valRect.Width = widths[n];
            }
            return checkedItems[n].Trim();
        }

        private int GetItemIndex(CheckedComboBoxEdit edit, string description)
        {
            CheckedListBoxItemCollection collection = edit.Properties.GetItems();
            for(int i = 0; i < collection.Count; i++)
            {
                if(collection[i].Description.Equals(description))
                    return i;
            }
            return -1;
        }

        private int GetSeparatorWidth(Font font, char separatorChar, Graphics gr)
        {
            return DevExpress.Utils.Text.TextUtils.GetStringSize(gr,
                separatorChar.ToString(), font).Width;
        }

        private SuperToolTip GetSuperToolTip(CheckedComboBoxEdit edit, DataRow row)
        {
            SuperToolTip superToolTip = new SuperToolTip();
            ToolTipItem itemDecsr = new ToolTipItem();
            itemDecsr.Text = row[edit.Properties.DisplayMember].ToString();
            ToolTipItem itemValue = new ToolTipItem();
            itemValue.Text = row[edit.Properties.ValueMember].ToString();
            ToolTipItem itemImage = new ToolTipItem();
            itemImage.Image = GetToolTipImage(row, itemImage);
            superToolTip.Items.Add(itemDecsr);
            superToolTip.Items.Add(itemValue);
            superToolTip.Items.Add(itemImage);
            return superToolTip;
        }

        private Image GetToolTipImage(DataRow row, ToolTipItem itemImage)
        {
            using(MemoryStream stream = new MemoryStream((byte[])row["Picture"]))
                return Image.FromStream(stream);
        }

        private int[] GetValuesWidths(Font font, string[] values)
        {
            GraphicsInfo info = new GraphicsInfo();
            info.AddGraphics(null);

            int[] widths = new int[values.Length];
            int separatorWidth = GetSeparatorWidth(font, separatorChar, info.Graphics);
            for(int i = 0; i < values.Length - 1; i++)
                widths.SetValue(DevExpress.Utils.Text.TextUtils.GetStringSize(info.Graphics,
                    values[i], font).Width + separatorWidth, i);
            widths.SetValue(DevExpress.Utils.Text.TextUtils.GetStringSize(info.Graphics,
                    values[values.Length - 1], font).Width, values.Length - 1);

            info.ReleaseGraphics();

            return widths;
        }

        private void OnGetActiveObject(object sender, DevExpress.Utils.ToolTipControllerGetActiveObjectInfoEventArgs e)
        {
            if(e.SelectedControl == checkedComboBoxEdit1)
            {
                PopupContainerEditViewInfo editInfo = checkedComboBoxEdit1.GetViewInfo() as PopupContainerEditViewInfo;
                EditHitInfo hitInfo = editInfo.CalcHitInfo(e.ControlMousePosition);
                if(hitInfo.HitTest == EditHitTest.MaskBox)
                {
                    object val = checkedComboBoxEdit1.EditValue;
                    if(val == null || val == DBNull.Value || val.Equals(string.Empty)) return;
                    string[] checkedItems = checkedComboBoxEdit1.Text.Split(separatorChar);
                    Rectangle rect = editInfo.MaskBoxRect;
                    string description = GetCheckedItemDescription(checkedItems, editInfo.PaintAppearance.Font, e.ControlMousePosition, rect);
                    int itemIndex = GetItemIndex(checkedComboBoxEdit1, description);
                    if(itemIndex == -1) return;
                    ToolTipControlInfo tInfo = tInfo = new ToolTipControlInfo();
                    tInfo.Object = itemIndex;
                    tInfo.SuperTip = GetSuperToolTip(checkedComboBoxEdit1, (carsBindingSource.DataSource as DataTable).Rows[itemIndex]);
                    e.Info = tInfo;
                }
            }
        }
    }

}

File: VB/WindowsApplication3/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.Utils.Drawing
Imports DevExpress.XtraGrid.Views.Layout
Imports DevExpress.Utils
Imports DevExpress.XtraEditors.Controls
Imports System.IO


Namespace WindowsApplication3
	Partial Public Class Form1
		Inherits XtraForm
		Public Sub New()
			InitializeComponent()
		End Sub

		Private separatorChar As Char

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			' TODO: This line of code loads data into the 'carsDBDataSet.Cars' table. You can move, or remove it, as needed.
			Me.carsTableAdapter.Fill(Me.carsDBDataSet.Cars)
			separatorChar = checkedComboBoxEdit1.Properties.SeparatorChar
		End Sub

		Private Sub OnGetActiveObject(ByVal sender As Object, ByVal e As DevExpress.Utils.ToolTipControllerGetActiveObjectInfoEventArgs) Handles toolTipController1.GetActiveObjectInfo
			If e.SelectedControl Is checkedComboBoxEdit1 Then
				Dim editInfo As PopupContainerEditViewInfo = TryCast(checkedComboBoxEdit1.GetViewInfo(), PopupContainerEditViewInfo)
				Dim hitInfo As EditHitInfo = editInfo.CalcHitInfo(e.ControlMousePosition)
				If hitInfo.HitTest = EditHitTest.MaskBox Then
					Dim val As Object = checkedComboBoxEdit1.EditValue
					If val Is Nothing OrElse val Is DBNull.Value OrElse val.Equals(String.Empty) Then
						Return
					End If
					Dim checkedItems() As String = checkedComboBoxEdit1.Text.Split(separatorChar)
					Dim rect As Rectangle = editInfo.MaskBoxRect
					Dim description As String = GetCheckedItemDescription(checkedItems, editInfo.PaintAppearance.Font, e.ControlMousePosition, rect)
					Dim itemIndex As Integer = GetItemIndex(checkedComboBoxEdit1, description)
					If itemIndex = -1 Then
						Return
					End If
                    Dim tInfo As ToolTipControlInfo = New ToolTipControlInfo()
					tInfo.Object = itemIndex
					tInfo.SuperTip = GetSuperToolTip(checkedComboBoxEdit1, carsDBDataSet.Cars(itemIndex))
					e.Info = tInfo
				End If
			End If
		End Sub

		Private Function GetItemIndex(ByVal edit As CheckedComboBoxEdit, ByVal description As String) As Integer
			Dim collection As CheckedListBoxItemCollection = edit.Properties.GetItems()
			For i As Integer = 0 To collection.Count - 1
				If collection(i).Description.Equals(description) Then
					Return i
				End If
			Next i
			Return -1
		End Function

		Private Function GetSuperToolTip(ByVal edit As CheckedComboBoxEdit, ByVal row As DataRow) As SuperToolTip
			Dim superToolTip As New SuperToolTip()
			Dim itemDecsr As New ToolTipItem()
			itemDecsr.Text = row(edit.Properties.DisplayMember).ToString()
			Dim itemValue As New ToolTipItem()
			itemValue.Text = row(edit.Properties.ValueMember).ToString()
			Dim itemImage As New ToolTipItem()
			itemImage.Image = GetToolTipImage(row, itemImage)
			superToolTip.Items.Add(itemDecsr)
			superToolTip.Items.Add(itemValue)
			superToolTip.Items.Add(itemImage)
			Return superToolTip
		End Function

		Private Function GetToolTipImage(ByVal row As DataRow, ByVal itemImage As ToolTipItem) As Image
			Using stream As New MemoryStream(CType(row("Picture"), Byte()))
				Return Image.FromStream(stream)
			End Using
		End Function

		Private Function GetCheckedItemDescription(ByVal checkedItems() As String, ByVal font As Font, ByVal point As Point, ByVal rect As Rectangle) As String
			Dim widths() As Integer = GetValuesWidths(font, checkedItems)
			Dim valRect As New Rectangle(rect.X, rect.Y, widths(0), rect.Height)
			Dim n As Integer = 0
			Do While (Not valRect.Contains(point)) AndAlso n < widths.Length - 1
				valRect.X += widths(n)
				n += 1
				valRect.Width = widths(n)
			Loop
			Return checkedItems(n).Trim()
		End Function

		Private Function GetValuesWidths(ByVal font As Font, ByVal values() As String) As Integer()
			Dim info As New GraphicsInfo()
			info.AddGraphics(Nothing)

			Dim widths(values.Length - 1) As Integer
			Dim separatorWidth As Integer = GetSeparatorWidth(font, separatorChar, info.Graphics)
			For i As Integer = 0 To values.Length - 2
				widths.SetValue(DevExpress.Utils.Text.TextUtils.GetStringSize(info.Graphics, values(i), font).Width + separatorWidth, i)
			Next i
			widths.SetValue(DevExpress.Utils.Text.TextUtils.GetStringSize(info.Graphics, values(values.Length - 1), font).Width, values.Length - 1)

			info.ReleaseGraphics()

			Return widths
		End Function

		Private Function GetSeparatorWidth(ByVal font As Font, ByVal separatorChar As Char, ByVal gr As Graphics) As Integer
			Return DevExpress.Utils.Text.TextUtils.GetStringSize(gr, separatorChar.ToString(), font).Width
		End Function
	End Class

End Namespace